bundles/EcommerceFrameworkBundle/src/Environment.php

```diff
@@ -33,7 +33,7 @@ class Environment implements EnvironmentInterface
 
     protected int $userId = self::USER_ID_NOT_SET;
 
-    protected bool $useGuestCart = false;
+    protected ?bool $useGuestCart = null;
```

Comment:
Made it nullable for integrating App override into core..https://github.com/pimcore/demo/blob/11.x/src/Ecommerce/Overrides/Environment.php#L19